src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,727 @@
+// separators     = "(" | ")" | "<" | ">" | "@"
+//                  | "," | ";" | ":" | "\" | <">
+//                  | "/" | "[" | "]" | "?" | "="
+//                  | "{" | "}" | SP | HT
+// TEXT           = <any OCTET except CTLs, but including LWS>
+// OCTET          = <any 8-bit sequence of data>
+// CHAR           = <any US-ASCII character (octets 0 - 127)>
+// CTL            = <any US-ASCII control character(octets 0 - 31)and DEL(127)>
+// CR             = <US-ASCII CR, carriage return (13)>
+// LF             = <US-ASCII LF, linefeed (10)>
+// SP             = <US-ASCII SP, space (32)>
+// HT             = <US-ASCII HT, horizontal-tab (9)>
+// <">            = <US-ASCII double-quote mark (34)>
+// LWS            = [CRLF] 1*( SP | HT )
+// CRLF           = CR LF
+// Linear white space (LWS) MUST NOT be used between the type and subtype, nor 
+// between an attribute and its value. Leading and trailing LWS are prohibited.
+// 
+//-----------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;   // For Dictionary<string, string>
+using System.Text;                  // For StringBuilder
+using System.Diagnostics;           // For Debug.Assert
+
+namespace System.IO.Packaging
+{
+    /// <summary>
+    /// Content Type class
+    /// </summary>
+    internal sealed class ContentType
+    {
+        //------------------------------------------------------
+        //
+        //  Internal Constructors
+        //
+        //------------------------------------------------------
+
+        #region Internal Constructors
+
+        /// <summary>
+        /// This constructor creates a ContentType object that represents 
+        /// the content-type string. At construction time we validate the 
+        /// string as per the grammar specified in RFC 2616.
+        /// Note: We allow empty strings as valid input. Empty string should
+        /// we used more as an indication of an absent/unknown ContentType. 
+        /// </summary>
+        /// <param name="contentType">content-type</param>
+        /// <exception cref="ArgumentNullException">If the contentType parameter is null</exception>
+        /// <exception cref="ArgumentException">If the contentType string has leading or 
+        /// trailing Linear White Spaces(LWS) characters</exception>
+        /// <exception cref="ArgumentException">If the contentType string invalid CR-LF characters</exception>
+        internal ContentType(string contentType)
+        {
+            if (contentType == null)
+                throw new ArgumentNullException("contentType");
+
+            if (String.CompareOrdinal(contentType, String.Empty) == 0)
+            {
+                _contentType = String.Empty;
+            }
+            else
+            {
+                if (IsLinearWhiteSpaceChar(contentType[0]) || IsLinearWhiteSpaceChar(contentType[contentType.Length - 1]))
```

Comment:
It doesn't appear that we know at this point that contentType isn't empty.  Couldn't this result in indexing out of bounds into contentType?